src/inventory_ui.cpp

```diff
@@ -884,9 +892,71 @@ void inventory_column::prepare_paging( const std::string &filter )
             from = to;
         }
     }
+
+    if( !grouped ) {
+        // remove group entries
+        for( const std::shared_ptr< group_entry > &g_iter : group_entries_ ) {
+            // TODO: child entries need to be put back!
+            std::remove( entries.begin(), entries.end(), *g_iter );
+        }
+        group_entries_.clear();
+
+        std::vector<item_location> group;
+        bool inside_group = false;
+        std::vector<inventory_entry>::iterator it, nx, gp;
+        it = nx = gp = entries.begin();
+        while( it != entries.end() ) {
+            nx++;
+            if( nx != entries.end() && it->is_item() && nx->is_item() &&
+                it->any_item()->typeId() == nx->any_item()->typeId() ) {
+                // save items
+                group.insert( group.end(), it->locations.begin(), it->locations.end() );
+                if( !inside_group ) {
+                    gp = it;
+                    inside_group = true;
+                }
+            } else if( inside_group ) {
+                // save items
+                group.insert( group.end(), it->locations.begin(), it->locations.end() );
+                // create group entry
+                int children = std::distance( gp, nx );
+                group_entry g_entry( group, it->get_category_ptr() );
+                g_entry.update_cache();
+                g_entry.highlight_as_parent = true;
+                g_entry.set_number_of_children( children );
+                group_entries_.push_back(  // cant't be in main scope?
+                    std::make_shared<group_entry> ( g_entry ) );
+
+                nx = entries.insert( gp, g_entry );
```

Comment:
I don't think this will work as intended. `g_entry` will likely just be up-casted to an `inventory_entry` and added to the list, removing any `group_entry` specific members in the process.

Reply:
Yeah, but if I'm adding it to the list why would I cast it? The list is `protected`, adding and removing from it is controlled. So, yes overrides in `selection_column` will need to be checked, but other than that I'd prefer inheritance over extension.

Reply:
> why would I cast it

Maybe there's some confusion, but I was not suggesting that _you_ should cast it. `g_entry` is _implicitly_ cast in your code, and will only behave like a non-inherited `inventory_entry` in the list, which might not be what you intended.

Reply:
I get it, but I use `group_entry` only to insert it to `inventory_column::entries`, and that's only exposed to derived classes.

Reply:
If it's intended, an explicit cast and a comment explaining it would be nice.